scipy/optimize/tests/test_optimize.py

```diff
@@ -502,6 +502,18 @@ def test_ncg_hessp(self):
                         atol=1e-6, rtol=1e-7)
 
 
+def test_maxfev_test():
+    def cost(x):
+        return np.random.rand(1) * 1000  # never converged problem
```

Comment:
Nice idea, but making it deterministic for debugging might be handy.
```suggestion
def test_maxfev_test():
    rng = np.random.default_rng(271707100830272976862395227613146332411)
    def cost(x):
        return rng.random(1) * 1000  # never converged problem
```